### PR TITLE
feat: update milestone logic to slot

### LINKED ISSUE #7854: [Task]: Update milestone logic to slot
### Task description

- The `Unix Time` field of the Timelock Unlock Condition is replaced by `Slot Index`.
- The `Unix Time` field of the Expiration Unlock Condition is replaced by `Slot Index`.
- Update node metrics
- References to `milestone`

### LINKED ISSUE #7891: [Task]: Update `Timelock Unlock Condition` from `unix` to `slot`
### Task description

https://github.com/iotaledger/tips/blob/tip38/tips/TIP-0038/tip-0038.md#timelock-unlock-condition

### LINKED ISSUE #7892: [Task]: Update `Expiration Unlock Condition` from `unix` to `slot`
### Task description

https://github.com/iotaledger/tips/blob/tip38/tips/TIP-0038/tip-0038.md#expiration-unlock-condition

### PR DESCRIPTION
**Pull request by cpl121**
Closes #7854
Closes #7891
Closes #7892 

[Reference](https://github.com/iotaledger/tips/blob/tip46/tips/TIP-0046/tip-0046.md#slot-index)

## Summary

> Please summarize your changes, describing **what** they are and **why** they were made.

...

## Changelog

```
- Please write a list of changes
```

### Platforms

> Please select any platforms where your changes have been tested.

-   **Desktop**
    -   [ ] MacOS
    -   [ ] Linux
    -   [ ] Windows

### Instructions

> Please describe the specific instructions, configurations, and/or test cases necessary to **test and verify** that your changes work as intended.

...

## Checklist

> Please tick the following boxes that are relevant to your changes.

-   [ ] I have followed the contribution guidelines for this project
-   [ ] I have performed a self-review of my own code
-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have added or modified tests that prove my changes work as intended
-   [ ] I have verified that new and existing unit tests pass locally with my changes
-   [ ] I have verified that my latest changes pass CI workflows for testing and linting
-   [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/shared/lib/core/wallet/utils/transactions/getSenderAddressFromInputs.ts [hunk 1]
```diff
@@ -14,14 +14,18 @@ export function getSenderAddressFromInputs(inputs: IWrappedOutput[]): string | u
         const { output, metadata } = input
         const unlockConditions = (output as CommonOutput)?.unlockConditions
 
-        const spentDate = metadata?.milestoneTimestampSpent
+        // TODO(2.0): modify this
```
Comment (evavirseda) [inline packages/shared/lib/core/wallet/utils/transactions/getSenderAddressFromInputs.ts:17]: perhaps we should create an issue for this and add it to the backlog, just to make sure we don't forget about it, wdyt? :pray:

File: packages/shared/lib/contexts/governance/utils/isVotingForProposal.ts [hunk 2]
```diff
@@ -3,6 +3,7 @@ import type { TrackedParticipationOverview } from '@iota/sdk/out/types'
 import { getParticipationsForProposal } from './getParticipationsForProposal'
 import { selectedWalletId } from '@core/wallet'
 
+// TODO(2.0): Update milestone by slot when TrackedParticipationOverview has been updated
```
Comment (evavirseda) [inline packages/shared/lib/contexts/governance/utils/isVotingForProposal.ts:6]: same here

File: packages/shared/lib/contexts/governance/utils/createProposalFromEvent.ts [hunk 3]
```diff
@@ -12,7 +12,8 @@ export function createProposalFromEvent(event: ParticipationEventWithNodes): IPr
     const nodeUrl = event.nodes[0].url
     const isOfficialNetwork = officialNodeUrls.includes(nodeUrl)
 
-    const milestones = {
+    // TODO(2.0): Update ProposalStatus when ParticipationEventData has been updated
```
Comment (evavirseda) [inline packages/shared/lib/contexts/governance/utils/createProposalFromEvent.ts:15]: same

File: packages/desktop/views/dashboard/governance/views/ProposalDetailsView.svelte [hunk 4]
```diff
@@ -121,6 +121,7 @@
         }
     }
 
+    // TODO(2.0): Update ProposalStatus when ParticipationEventData and TrackedParticipationOverview has been updated
```
Comment (evavirseda) [inline packages/desktop/views/dashboard/governance/views/ProposalDetailsView.svelte:124]: same

**Review by evavirseda — APPROVED** [review]
nice job, just a few suggestions :pray:

File: packages/shared/lib/core/wallet/utils/transactions/getSenderAddressFromInputs.ts (hunk 1 above)
Comment (cpl121) [reply packages/shared/lib/core/wallet/utils/transactions/getSenderAddressFromInputs.ts]: Modified! ✅

File: packages/shared/lib/contexts/governance/utils/isVotingForProposal.ts (hunk 2 above)
Comment (cpl121) [reply packages/shared/lib/contexts/governance/utils/isVotingForProposal.ts:6]: https://github.com/iotaledger/firefly/issues/7953

**Pull request merged**